lib/puppetserver/ca/puppet_config.rb

```diff
@@ -75,19 +74,28 @@ def resolve_settings(overrides = {})
         cadir = overrides.fetch(:cadir, '$ssldir/ca')
         settings[:cadir] = substitutions['$cadir'] = cadir.sub(unresolved_setting, substitutions)
 
-        settings[:cacert] = overrides.fetch(:cacert, '$cadir/ca_crt.pem')
-        settings[:cakey] = overrides.fetch(:cakey, '$cadir/ca_key.pem')
-        settings[:cacrl] = overrides.fetch(:cacrl, '$cadir/ca_crl.pem')
-        settings[:serial] = overrides.fetch(:serial, '$cadir/serial')
+        # Note the call to hostname below, (tbd from Justin's code)
+        settings[:certname] = substitutions['$certname'] = overrides.fetch(:certname, `hostname`.chomp)
+
+        settings[:ca_name] =        overrides.fetch(:ca_name, 'Puppet CA: $certname')
+        settings[:root_ca_name] =   overrides.fetch(:root_ca_name, 'Puppet Root CA: $certname')
+        settings[:ca_ttl] =         overrides.fetch(:ca_ttl, '15y')
```

Comment:
What if we called the TTLSetting class here and added any errors that it finds to this classes errors?